Set up Mongoose and Created User Schema

melvyn9 opened the pull request:
1. Connected to the mongoDB server using mongoose. Removed database.js as it was not needed.
2. Created user schema.

melvyn9 committed:
Set up Mongoose and Created User Schema

1. Connected to the mongoDB server using mongoose. Removed database.js as it was not needed.
2. Created user schema.

melvyn9 committed:
Added export to User Schema

Exported User for use in other files.

shad674 commented on models/user.js:
```diff
@@ -0,0 +1,30 @@
+    email: {
+        type: String,
+        required: true
+    }
```
This is a good start! We will add more schema in the future in terms of the people they like to be room with!

shad674 commented on models/user.js:
```diff
@@ -0,0 +1,30 @@
+const Schema = mongoose.Schema; // Used to create a new Schema
+
+// Define the Schema for user
+// TODO we can check for validation in the future
```
For validation and the next CR can you connect one of the GET endpoint to get any data from the database

melvyn9 replied:
Resolved

melvyn9 committed:
Read and Write Capabilities

1. Data can be read at /users.
2. Data can be added through Postman at /add-user
3. Added comments.

shad674 commented on controllers/admin.js:
```diff
@@ -0,0 +1,29 @@
+exports.addUser = (req, res, next) => {
+    // Obtain data and store it.
+    const firstName = req.body.firstName;
+    const lastName = req.body.lastName;
```
Nice

shad674 commented on server.js:
```diff
@@ -1,15 +1,57 @@
+app.use(mainRoutes);
+
+// Mongoose
+const mongoose = require('mongoose');
```
Nice formatting

shad674 reviewed (approved):
LGTM
